File: classes/parte-6-datos-y-estructuras/104-archivos-leer-y-escribir-texto-y-binario/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	linea := strings.TrimRight(line, "\r\n")
	palabras := len(strings.Fields(linea))
	fmt.Printf("palabras=%d caracteres=%d\n", palabras, len(linea))
}

File: classes/parte-6-datos-y-estructuras/104-archivos-leer-y-escribir-texto-y-binario/implementaciones/rust/main.rs
use std::io::Read;

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let linea = s.trim_end_matches(['\r', '\n']);
    let palabras = linea.split_whitespace().count();
    println!("palabras={} caracteres={}", palabras, linea.len());
}
